modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -908,6 +909,32 @@ public static <T> List<T> collectStaticFields(Class<?> sourceCls, Class<? extend
         return result;
     }
 
+    /**
+     * Cancels the future and runs a consumer on future's result if it was completed before the cancellation.
+     * Does nothing if future is cancelled or completed exceptionally.
+     *
+     * @param future Future.
+     * @param consumer Consumer that accepts future's result.
+     * @param <T> Future's result type.
+     */
+    public static <T> void cancelOrConsume(CompletableFuture<T> future, Consumer<T> consumer) {
+        assert future != null;
```

Comment:
Why do you need this assertion?

Reply:
ok, how about `@Contract("null -> fail")`?

Reply:
why?

Reply:
So that idea would highlight that null is not an option here

Reply:
It should highlight that anyway, because this parameter is not marked as `@Nullable`